legion/engine/physics/data/colliders/collider_data.hpp

```diff
@@ -33,23 +33,25 @@ namespace legion::physics
         collider_type m_colliderType = collider_type::not_set;
     };
 
+    using InternalConvexColliderPtr = void*;
+
     class ConvexColliderData : public ColliderData
     {
     public:
         friend class ConvexColliderHandle;
 
         ConvexColliderData(const math::vec3& offset,const math::quat& rotation, const math::vec3& boxExtents) noexcept;
 
-        ConvexColliderData(const math::vec3& offset, const math::quat& rotation, void* internalConvexColliderStructure);
+        ConvexColliderData(const math::vec3& offset, const math::quat& rotation, const InternalConvexColliderPtr ccPtr);
 
         L_ALWAYS_INLINE const math::vec3& getBoxExtents() const noexcept { return m_boxExtents; }
 
-        L_ALWAYS_INLINE pointer<void> getConvexPtr() const noexcept { return { m_internalConvexStructure }; }
+        L_ALWAYS_INLINE pointer<const void> getConvexPtr() const noexcept { return { m_internalConvexStructure }; }
```

Comment:
little bit of inconsistent usage between pointer<T> and T*. not crucial, but a little bit confusing to a reader of the code